src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java

```diff
@@ -35,39 +35,95 @@
 
     private Evaluable q2 = NAN;
 
-    protected LfBranchImpl(LfBus bus1, LfBus bus2, PiModel piModel, Branch branch) {
+    protected LfBranchImpl(LfBus bus1, LfBus bus2, PiModel piModel, PhaseControl phaseControl, Branch branch) {
         super(bus1, bus2, piModel);
+        this.phaseControl = phaseControl;
         this.branch = branch;
     }
 
+    private static LfBranchImpl createLine(Line line, LfBus bus1, LfBus bus2, double zb) {
+        PiModel piModel = new SimplePiModel()
+                .setR(line.getR() / zb)
+                .setX(line.getX() / zb)
+                .setG1(line.getG1() * zb)
+                .setG2(line.getG2() * zb)
+                .setB1(line.getB1() * zb)
+                .setB2(line.getB2() * zb);
+
+        return new LfBranchImpl(bus1, bus2, piModel, null, line);
+    }
+
+    private static LfBranchImpl createTransformer(TwoWindingsTransformer twt, LfBus bus1, LfBus bus2, double nominalV1,
+                                                  double nominalV2, double zb) {
+        PiModel piModel;
+        PhaseControl phaseControl = null;
+
+        PhaseTapChanger ptc = twt.getPhaseTapChanger();
+        if (ptc != null
+                && ptc.isRegulating()
+                && ptc.getRegulationMode() != PhaseTapChanger.RegulationMode.FIXED_TAP) {
+            PhaseTapChanger.RegulationMode regulationMode = ptc.getRegulationMode();
+            PhaseControl.ControlledSide controlledSide;
+            if (ptc.getRegulationTerminal() == twt.getTerminal1()) {
+                controlledSide = PhaseControl.ControlledSide.ONE;
+            } else if (ptc.getRegulationTerminal() == twt.getTerminal2()) {
+                controlledSide = PhaseControl.ControlledSide.TWO;
```

Comment:
I am not sure to understand : if PTC are always located on side 1 in IIDM, where do we put the PTC in the LFBranch ? If it is also always side 1 (remember for example that in the Pi Model, R2 is set to zero), do we really support a A1 variable and a side 2 for the regulating terminal ?

Reply:
The side of the phase shift in the PI model is a different thing from the controlled side which will be later to allowed to not only be side 1 ou 2 of the phase shifter but another remote terminal.

Reply:
Ok.